l10n_nl_xaf_auditfile_export/models/xaf_auditfile_export.py

```diff
@@ -252,39 +252,55 @@ def get_taxes(self):
         ])
 
     @api.multi
-    def get_ob_totals(self):
-        """return totals of opening balance"""
-        self.env.cr.execute(
-            "select sum(l.credit), sum(l.debit), count(distinct a.id) "
+    def get_ob_sql_fragment(self):
+        """
+        return a from clause and a where clause selecting the
+        relevant lines for the opening balance
+        """
+        return self.env.cr.mogrify(
             "from account_move_line l, account_account a, "
-            "     account_account_type t "
+            "     account_move m, account_account_type t "
             "where a.user_type_id = t.id "
-            "and l.account_id = a.id "
-            "and l.date < %s "
+            "and a.id = l.account_id and l.date < %s "
+            "and l.move_id = m.id and m.state = 'posted' "
             "and l.company_id=%s "
             "and t.include_initial_balance = true ",
             (self.date_start, self.company_id.id),
+        ).decode('utf8')
+
+    @api.multi
+    def get_ob_totals(self):
+        """return totals of opening balance"""
+        self.env.cr.execute(
+            # pylint: disable=sql-injection
+            "select sum(l.balance), count(distinct a.id) " +
+            self.get_ob_sql_fragment() +
+            "and l.balance < 0"
```

Comment:
good question, I did this to be consistent with https://github.com/OCA/l10n-netherlands/pull/411/files#diff-f022b656919333e9fc10d33f53e72e0de1db9cd6722667987e4367d6f8d9afc8R130 (make it a credit if balance is negative, debit otherwise) - I don't feel qualified to decide if that's correct

Reply:
oh wait, I answered the wrong question. But am in the process of rewriting this as of the above anyways, so let's stash this for now

Reply:
It's not about the sign, the sign is alright - I'm talking more of the where vs having thing.

If we state the concise specification of what we wanted to achieve here as Stefan's "... group the balances retrieved in get_ob_lines by sign (debit/credit) and sum them ..." then that's not what you're doing here - here you're summing all negative move line balances and all positive move line balances per account. If you use the `having` notation instead, you'd be summing all balances first, and then using the sign of the result to determine if the sum counts as a credit or a debit. The latter is what I thought we agreed we should do.

Reply:
... and what your last push changes the code to indeed do.

Reply:
hmm, wouldn't that mean there's always one of totalCredit or totalDebit 0?

My understanding was we want

totalCredit = sum(//openingBalance/obLine/amnt[../amntTp/text()='C'])
totalDebit = sum(//openingBalance/obLine/amnt[../amntTp/text()='D'])

if this xpath makes sense?

Reply:
Well yes, your xpath is what I meant also. So I think we're in agreement now.